### ISSUE TITLE
56.2.0 20180514223240 can not re-enable extensions that require Firefox 57 or greater

### ISSUE THREAD

**Issue by grahamperrin** [BrowserWorks/waterfox#560]
First run after installing 56.2.0 build 20180514223240 over 56.1.0.111172 on FreeBSD-CURRENT: 

![2018-05-15 02 52 52 waterfox update](https://user-images.githubusercontent.com/192271/40032613-bf2f4444-57ec-11e8-86f1-b80f22d317ba.png)

![2018-05-15 02 53 15 waterfox update](https://user-images.githubusercontent.com/192271/40032618-c3887a74-57ec-11e8-99ba-4b9353410bd1.png)

![2018-05-15 02 53 37 waterfox update](https://user-images.githubusercontent.com/192271/40032623-c76c9a9e-57ec-11e8-9d4c-c6216996ef6c.png)

![2018-05-15 02 55 07 add-ons manager](https://user-images.githubusercontent.com/192271/40032663-f35ee1e8-57ec-11e8-9fef-8cc281a561cb.png)

… and so on, no **Enable** button for any of the five that are disabled …

![2018-05-15 02 58 57 add-ons manager](https://user-images.githubusercontent.com/192271/40032695-10a1f6a0-57ed-11e8-903c-9b85ac15bc01.png)

I wondered whether [Simple Add-on Manager](https://addons.mozilla.org/addon/simple-add-on-manager/) would allow an override. Unfortunately not: 

![2018-05-15 03 00 04 group -1 - speed dial](https://user-images.githubusercontent.com/192271/40032733-4a1a6a2a-57ed-11e8-835e-d4c00c6d181f.png)

For some of the extensions: 

- a downgrade will allow use, but (at least with Mobile Dyslexic) downgrading will lose functionality that was present with 56.1.0. 

For the others: 

- no legacy version is available.

**Comment by MrAlex94** [comment]
Ahh, can you add the pref: extensions.checkCompatibility.56.2;false and see
if that works?

**Comment by grahamperrin** [comment]
Perfect, thanks. I guessed that there would be a compatibility check workaround in prefs, but I was too lazy to figure it out for myself :-)

Result: 

- all five extensions automatically enabled immediately after adding the preference (none were disabled by me before the installation of 56.2.0)
- at about:addons the yellow alerts remain – 

> ⚠ Tip Tab is incompatible with Waterfox 56.2.0

– and so on. The alerts are proper, <strike>end </strike>power users may choose to ignore them.

**Comment by grahamperrin** [comment]
Postscript: ignore this comment. Apology/explanation below at https://github.com/MrAlex94/Waterfox/issues/560#issuecomment-389393793

----

Thoughts … for now (for the forthcoming general release), is it possible to include `extensions.checkCompatibility.56.2` `true` by default but `false` on first run of 56.2.0?

Rationale: 

- it **is** the type of check that should be made when refreshing 56.2.0. 

For 'power users' we can have, amongst FAQ, a shortlist of preferences such as this.

**Comment by MrAlex94** [comment]
Well, anyone already on 56.1 will have the same compatibility at 56.2, for
example. The compatibility check won't really help will it?

**Comment by grahamperrin** [comment]
From the POV of me troubleshooting other users' problems: I prefer caution. 

Strictly, **a refresh** – as part of a troubleshooting routine – should unambiguously **decrease** the risk of problems. 

----

For myself (YMMV) I have barely touched upon extensions that require Firefox 60 or greater. Touch wood, I haven't seen anything that might be troublesome (I see some features of an extension simply not work, but I wouldn't class that as 'trouble'). Fine for me :-) but touching wood, for luck, is not a methodical approach to troubleshooting. 

Can we _now_ be certain that _all_ extensions that require, or will require, Firefox 57, 58, 59, 60, 61, 62 … will be non-troublesome with Waterfox 56.2.x? Admittedly I don't know the code etc. but I should err on the side of caution. 

(The e10s stuff, too – https://github.com/MrAlex94/Waterfox/issues/397#issuecomment-370620417 – I wince every time I think of someone performing a refresh routine that might worsen the situation.)

**Comment by grahamperrin** [comment]
#538

**Comment by Peacock365** [comment]
@grahamperrin 

Well, you should realize that some extensions might be breaking when you force-enable them, because they might require Firefox 57+ WebExtension APIs that might not be present in Waterfox 56 at all. @Happy-Ferret has offered his help in #389, we could try to come back to this dev.

**Comment by grahamperrin** [comment]
Sorry, my <https://github.com/MrAlex94/Waterfox/issues/560#issuecomment-389029400> above was a brain fart. Critically: 

- I forgot that the effect of the boolean is _not_ limited to extensions that are supposedly 'too high' for Waterfox

– there's also the very well established 'power' use case of Waterfox being supposedly too high for some legacy extensions, so (unless I'm having a second bf) `false` is the sane default. 

----

`extensions.checkCompatibility.56.2` is present and `false` by default in 56.2.0.7 (56.2.0 build 20180515213058) on FreeBSD-CURRENT, so I guess that this issue can close after 56.2.0 goes to general release for other platforms.

**Comment by grahamperrin** [comment]
From <https://www.reddit.com/comments/8k48rk/-/dz50k5m/> three hours ago: 

> I did commit it, but seems I forgot to sync the final patch before building and releasing everything 🤦‍♂️.

Hey, these things happen :-)

**Comment by grahamperrin** [comment]
## about:config?filter=extensions.checkCompatibility.56.2

I found `extensions.checkCompatibility.56.2` present and `false` by default following a first run of build 20180515140315 – on an outdated pre-release of High Sierra – that resulted in: 

- [Waterfox 56.2.0 on Mac OS X launching with an untitled dialogue with four buttons alone – Cancel | Go Back | Continue | Done · Issue #572 …](https://github.com/MrAlex94/Waterfox/issues/572)

**Comment by AncalagonX** [comment]
I upgraded to Waterfox 56.2.0 today, and also finally upgraded my RES (Reddit Enhancement Suite) to the current version released last week. I believe I experienced the same problem the original issue poster had, where RES was spuriously set to "disabled" due to incompatibility. The simple solution posted above by @MrAlex94 worked perfectly for me:

Adding the boolean ```extensions.checkCompatibility.56.2``` and setting it to ```false``` fixed this error, and allowed me to enable the current Firefox RES version released last week (actually, as soon as I added that boolean, the RES addon it auto-enabled itself). Thanks for this solution, and thanks for all the hard work from @MrAlex94 and other contributors on these releases!
